sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -315,6 +315,17 @@ case class DescribeRelation(
   override def output: Seq[Attribute] = DescribeTableSchema.describeTableAttributes()
 }
 
+/**
+ * The logical plan of the DESCRIBE relation_name col_name command that works for v2 tables.
+ */
+case class DescribeColumn(
+    relation: LogicalPlan,
+    colNameParts: Seq[String],
```

Comment:
is it possible that the column name is qualified like `catalog_name.ns1.tbl1.col`? Does Hive support qualified column names in `DESCRIBE`?

Reply:
A simple idea is to put an `UnresolvedAttribute` here, and analyzer can do the work for us.

Reply:
> Does Hive support qualified column names in `DESCRIBE`?

Hive [doesn't support qualified name](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-Describe) for column names. For example:

```
DESCRIBE hivesampletable default.hivesampletable.clientid

java.sql.SQLException: Error while processing statement: FAILED: Execution Error, return code 1 from org.apache.hadoop.hive.ql.exec.DDLTask. cannot find field default from [0:clientid, 1:querytime, 2:market, 3:deviceplatform, 4:devicemake, 5:devicemodel, 6:state, 7:country, 8:querydwelltime, 9:sessionid, 10:sessionpagevieworder]
```

Reply:
> A simple idea is to put an `UnresolvedAttribute` here, and analyzer can do the work for us.

Since we need to have the relation resolved first, we need to match like the following in the analyzer:
```scala
case DescribeColumn(r: ResolvedTable, u: UnresolvedAttribute, _) =>
...
```
Is that what you had in mind?

Reply:
If Hive doesn't support it, it's OK to follow Hive here. Let's make sure that other commands only allow column name without qualifies too.

Reply:
Sure, will take a look.

Reply:
> is it possible that the column name is qualified like `catalog_name.ns1.tbl1.col`? Does Hive support qualified column names in `DESCRIBE`?

@cloud-fan actually I didn't respond to your first question. Yes, Spark allows qualifying the column name (same in 2.4):
```
scala> sql("CREATE TABLE t (id bigint) USING csv")
scala> sql("describe t spark_catalog.default.t.id").show
+---------+----------+
|info_name|info_value|
+---------+----------+
| col_name|        id|
|data_type|    bigint|
|  comment|      NULL|
+---------+----------+
```
, whereas Hive does not support. I guess we cannot change the behavior at this point?

Reply:
Yea we can't break it. When we add v2 `DecribeColumExec` later, we may want to change `Seq[String]` to `Attribute` here, so that we can unify the column resolution logic for v1 and  v2 commands.